cmd/pod-scaler/admission.go

```diff
@@ -78,15 +93,151 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) admissio
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
 }
 
-func mutatePod(pod *corev1.Pod, build *buildv1.Build) {
+// mutatePodMetadata updates metadata labels for Pods created by Prow for rehearsals,
+// where default metadata points to the release repo instead of the repo under test.
+// We can fix this by updating to use the values from the configuration that the job
+// ends up running with.
+func mutatePodMetadata(pod *corev1.Pod) error {
```

Comment:
Do we really want to change the pod instead of only inferring that information for further processing?

Reply:
I'm not sure - I think mentally thinking about "take a Pod, map it's labels to metric fields, then process the metric fields just as if it were coming from Prometheus" is valuable to reduce mental overhead, and mutating the metadata seemed like a way to add some optional complexity on top without changing the flow, but I'm open to other ideas. I like that this continues to feed everything into a metadata label --> metric label conversion.

Reply:
It's not really that - the labels relate to the commits under test, those that get reported to, etc - they are not "wrong" in the sense that they continue to convey the same information on rehearsals as they do on all jobs. It's just in the case of rehearsals, we can pull data from another source for the requests